File: app/src/main/java/com/dabai/app/ui/components/RelationshipAvatar.kt
package com.dabai.app.ui.components
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.dabai.app.ui.theme.*
@Composable
fun RelationshipAvatar(relationship: String, nickname: String, size: Dp = 48.dp, fontSize: TextUnit = 20.sp) {
    val bgColor = when (relationship) {
        "爸爸", "父亲" -> RelationshipDad
        "妈妈", "母亲" -> RelationshipMom
        "爷爷", "外公" -> RelationshipGrandpa
        "奶奶", "外婆" -> RelationshipGrandma
        "哥哥", "弟弟" -> RelationshipBrother
        "姐姐", "妹妹" -> RelationshipSister
        else -> RelationshipDefault
    }
    val displayChar = nickname.take(1)
    Box(modifier = Modifier.size(size).clip(CircleShape).background(bgColor), contentAlignment = Alignment.Center) {
        Text(text = displayChar, fontSize = fontSize, fontWeight = FontWeight.Bold, color = Color.White)
    }
}

File: lib/widgets/relationship_avatar.dart
import "package:flutter/material.dart";
import "../config/constants.dart";

class RelationshipAvatar extends StatelessWidget {
  final String relationship;
  final String nickname;
  final double size;
  final double fontSize;

  const RelationshipAvatar({
    super.key,
    required this.relationship,
    required this.nickname,
    this.size = 48,
    this.fontSize = 20,
  });

  Color get backgroundColor {
    switch (relationship) {
      case "爸爸": case "父亲": return const Color(DabaiColors.relationshipDad);
      case "妈妈": case "母亲": return const Color(DabaiColors.relationshipMom);
      case "爷爷": case "外公": return const Color(DabaiColors.relationshipGrandpa);
      case "奶奶": case "外婆": return const Color(DabaiColors.relationshipGrandma);
      case "哥哥": case "弟弟": return const Color(DabaiColors.relationshipBrother);
      case "姐姐": case "妹妹": return const Color(DabaiColors.relationshipSister);
      default: return const Color(DabaiColors.relationshipDefault);
    }
  }

  @override
  Widget build(BuildContext context) {
    return Container(
      width: size, height: size,
      decoration: BoxDecoration(shape: BoxShape.circle, color: backgroundColor),
      child: Center(
        child: Text(
          nickname.isNotEmpty ? nickname.characters.first : "?",
          style: TextStyle(color: Colors.white, fontSize: fontSize, fontWeight: FontWeight.bold),
        ),
      ),
    );
  }
}
